weatherman.py

```diff
@@ -0,0 +1,398 @@
+
+def get_month_year_from_string(string):
+    month, year = string.split('/')
+    try:
+        month = int(month)
+        year = int(year)
+    except Exception:
+        raise "Invalid input"
+        return None, None
+    return month, year
+
+
+def average(lis):
+    sum_ = sum(lis)
+    return sum_ / len(lis)
+
+
+# Computes highest temperature, lowest temperature and humidity
+def minus_e(weather_data: WeatherData, year: int):
+    max_temp_list = []
+    min_temp_list = []
+    max_humidity_list = []
+    min_humidity_list = []
+
+    # Checking data of all 12 months
+    for i in range(0, 12):
+        month = i + 1
+        # Max temperature
+        col_name = weather_data.get_column_name_from_alias(
+                                "max_temperature")
+        if col_name is None:
+            print("Invalid column data")
+        else:
+            max_temp_column = weather_data.get_column_data(col_name,
+                                                           month, year)
+            if max_temp_column != []:
+                max_index = np.argmax(max_temp_column)
+                max_temp_list.append([max_temp_column[max_index],
+                                      max_index + 1, month])
+
+        # Min temperature
+        col_name = weather_data.get_column_name_from_alias(
+                                "min_temperature")
+        if col_name is None:
+            print("Invalid column alias")
+        else:
+            min_temp_column = weather_data.get_column_data(col_name,
+                                                           month, year)
+            if min_temp_column != []:
+                min_index = np.argmin(min_temp_column)
+                min_temp_list.append([min_temp_column[min_index],
+                                      min_index + 1, month])
+
+        # Max humidity
+        col_name = weather_data.get_column_name_from_alias(
+                                "max_humidity")
+        if col_name is None:
+            print("Invalid column alias")
+        else:
+            max_humidity_column = weather_data.get_column_data(col_name,
+                                                               month, year)
+            if max_humidity_column != []:
+                max_index = np.argmax(max_humidity_column)
+                max_humidity_list.append([max_humidity_column[max_index],
+                                          max_index + 1, month])
+
+        # Min humidity
+        col_name = weather_data.get_column_name_from_alias(
+                                "min_humidity")
+        if col_name is None:
+            print("Invalid column alias")
+        else:
+            min_humidity_column = weather_data.get_column_data(col_name,
+                                                               month, year)
+            if min_humidity_column != []:
+                min_index = np.argmin(min_humidity_column)
+                min_humidity_list.append([min_humidity_column[min_index],
```

Comment:
move all this logic for min humidity to a separate method, and call that method here. Would make code more cleaner.